.github/workflows/deploy_native.yml

```diff
@@ -0,0 +1,66 @@
+name: Build and Deploy Raspberry Pi OS Image
+
+on:
+  workflow_dispatch:
+  workflow_run:
+    workflows: ["Build and Deploy Images"]
+    types:
+      - completed
+
+jobs:
+  deploy-pi-img:
+    runs-on: self-hosted
+    # The runner for this job is a Raspberry Pi OS Bullseye.
+    # Just install docker (curl -sSL https://get.docker.com/ | sh) and follow the instructions for setting up a new runner in
+    # https://github.com/bluerobotics/BlueOS-docker/settings/actions/runners/new
+
+    strategy:
+      matrix:
+        platforms: ["linux/arm/v7"]
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+
+      -
+        name: Set env SHOULD_RUN
+        # sets a variabel that means we should run on branches/tags and manual dispatchs only. (no PRs)
```

Comment:
Can't you use only `github.event.issue.pull_request` ? and use directly inside the if configuration.
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment-on-issues-only-or-pull-requests-only